test/images/agnhost/webhook/main.go

```diff
@@ -54,22 +56,80 @@ func init() {
 		"File containing the default x509 private key matching --tls-cert-file.")
 }
 
-// toAdmissionResponse is a helper function to create an AdmissionResponse
-// with an embedded error
-func toAdmissionResponse(err error) *v1beta1.AdmissionResponse {
+// admitv1beta1Func handles a v1beta1 admission
+type admitv1beta1Func func(v1beta1.AdmissionReview) *v1beta1.AdmissionResponse
+
+// admitv1beta1Func handles a v1 admission
+type admitv1Func func(v1.AdmissionReview) *v1.AdmissionResponse
+
+// admitHandler is a handler, for both validators and mutators, that supports multiple admission review versions
+type admitHandler struct {
+	v1beta1 admitv1beta1Func
+	v1      admitv1Func
+}
+
+func newDelegateToV1AdmitHandler(f admitv1Func) admitHandler {
+	return admitHandler{
+		v1beta1: delegateV1beta1AdmitToV1(f),
+		v1:      f,
+	}
+}
+
+func delegateV1beta1AdmitToV1(f admitv1Func) admitv1beta1Func {
+	return func(review v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+		return convertAdmissionResponseTov1beta1(f(convertAdmissionReviewToV1(review)))
+	}
+}
+
+func convertAdmissionReviewToV1(r v1beta1.AdmissionReview) v1.AdmissionReview {
```

Comment:
would be good to have a unit test here that catches if a new field is introduced to AdmissionRequest or AdmissionResponse